File: SolvedProblems/MaxConsecutiveOnesIII/MaxConsecutiveOnesIII.js
const longestOnes = function (nums, k) {
  let startP = 0;
  let endP = 0;
  let zeroesCount = k;
  let windowSize = 0;

  while (endP < nums.length) {
    if (!nums[endP]) {
      zeroesCount--; // decrement zeroes count
    }

    if (zeroesCount < 0) {
      if (!nums[startP]) {
        // if startPoint equals 0
        zeroesCount++; // increment zeroes count
      }
      startP++; // move start Point to the next position
      windowSize--; // increment window size
    }

    endP++; // move end Point to the next position
    windowSize++; // incremement the window size
  }

  return windowSize;
};

longestOnes([0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1], 3);

File: SolvedProblems/MaxConsecutiveOnesIII/MaxConsecutiveOnesIII.py
def longestOnes(nums, k):
       startP = 0
       endP = 0
       zeroesCount = k

       while endP < len(nums):
            if nums[endP] == 0:
                zeroesCount-=1
            
            if zeroesCount < 0:
                if nums[startP] == 0:
                    zeroesCount+= 1
                startP+=1

            endP+=1

       return endP - startP

longestOnes([0,0,1,1,0,0,1,1,1,0,1,1,0,0,0,1,1,1,1], 3)